java-api-core/src/com/kinvey/java/network/NetworkManager.java

```diff
@@ -468,6 +462,18 @@ public String onUnsavedReferenceFound(String collection, GenericJson object) {
         return save;
     }
 
+    public boolean isTempId(T item) {
+        boolean isTempId = true;
+        if (item.get(Constants._ID) != null) {
+            try {
+                isTempId = item.get(Constants._ID).toString().matches("\\w{8}-\\w{4}-\\w{4}-\\w{4}-\\w{12}");
```

Comment:
Does this also have to check that the `_id` starts with `temp_`?

Reply:
Yes, this has to check that the `_id` starts with ` temp_`, but it was the feature from another branch which was merged in the current branch today, so currently the SDK already checks ` temp_`. If a user had items which weren't synced the items will be pushed using `PUT`, but all new items will be pushed with `POST` request. The PR has instrumented test with such case.